kqueue.go

```diff
@@ -119,6 +121,16 @@ func (w *Watcher) Remove(name string) error {
 	w.mu.Lock()
 	isDir := w.paths[watchfd].isDir
 	delete(w.watches, name)
+
+	parentName := filepath.Clean(filepath.Dir(name))
+	parentWatches := w.watchesByDir[parentName]
+	for i, fd := range parentWatches {
+		if fd == watchfd {
+			w.watchesByDir[parentName] = append(parentWatches[:i], parentWatches[i+1:]...)
```

Comment:
why don't we turn parentWatches into a `map[int]struct{}` or a `map[string]int`? then it'll be a lot easier to remove a child watch.

Reply:
done